### ISSUE TITLE
Rule Change: add maxLines option to arrow-body-style

### ISSUE THREAD

**Issue by JacobLey** [eslint/eslint#15814]
### What rule do you want to change?

arrow-body-style

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```ts
// Allowed
/*eslint arrow-body-style: ["error", "as-needed"]*/
const shortImplicitReturn = () => 123;
const shortImplicitFunctionReturn = () => foo(123);

// Allowed, but probably should not be allowed
/*eslint arrow-body-style: ["error", "as-needed"]*/
const grossLargeImplicitReturn = ( 
    { foo, bar }: { foo: string; bar: number },
    {
        destructured = true,
        optional = 123
    }: { 
        destructured?: boolean;
        optional?: number;
    } = {}
): Promise<Pick<CustomType, 'abc' | 'efg'>> => fetch(
    'https://example.com',
    {
        method: 'POST',
        body: {
            someData: true,
            foo,
            bar,
        },
        headers: {
            Authentication: 'secret',
            [optional]: destructured,
        },
    }
);

// Disallowed, but probably should be allowed
/*eslint arrow-body-style: ["error", "as-needed"]*/
const disAllowedExplicitReturn = ( 
    { foo, bar }: { foo: string; bar: number },
    {
        destructured = true,
        optional = 123
    }: { 
        destructured?: boolean;
        optional?: number;
    } = {}
): Promise<Pick<CustomType, 'abc' | 'efg'>> => {
    return fetch(
        'https://example.com',
        {
            method: 'POST',
            body: {
                someData: true,
                foo,
                bar,
            },
            headers: {
                Authentication: 'secret',
                [optional]: destructured,
            },
        }
    );
};
```

### What does the rule currently do for this code?

Currently the `as-needed` option for `arrow-body-style` enforces implicit return so long as the response is not an object literal (determined by `requireReturnForObjectLiteral`).

In some cases, the returned value is something like a function call that ends up being many lines long, and the result is harder to read. There is no option to "opt out" of implicit returns for those kinds of functions, except via `eslint-disable` (makes rule useless) or `never` (stops allowing implicit return for "simpler" functions)

### What will the rule do after it's changed?

In the case like `disAllowedExplicitReturn` above, I believe I would prefer to opt-out of forced implicit return based on the length of the function.

Some option such as `maxLines` could be used so that any functions with lines _greater_ than that will use explicit return. Similar to `requireReturnForObjectLiteral` it could be only an option for `as-needed`.

Potentially there could also be another rule such as `requireReturnForFunction` that forces explicit return when function immediately calls another function. Although in my example `shortImplicitFunctionReturn` I am fine with implicit return (but perhaps others would like the option)

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

If implemented, this should be a MINOR version bump.

The `maxLines` option would default to null (or Infinity?) so omitting would not change behavior.

`requireReturnForFunction` would default to false.

I am open to taking a swing at a PR for this. In my naive review, this seems pretty straightforward to enforce?

**Comment by mdjermanovic** [comment]
Hi @JacobLey, thanks for the issue! 

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we are not adding new options to the `arrow-body-style` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
